src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1109,29 +1059,33 @@ void Ekf2::run()
 					}
 				}
 
-				// Check and save the last valid calibration when we are disarmed
-				if ((vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_STANDBY)
-				    && (status.filter_fault_flags == 0)
+				// Check and update the last valid magnetometer calibration
```

Comment:
The EKF is already applying it's learned mag bias corrections internally, so applying them to the data continuously before it is consumed by the EKF and while the EKF runs will cause problems.